react-example/src/components/Arkose.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class Arkose extends React.Component {
+  constructor () {
+    super();
+    this.myEnforcement = null;
+  }
+
+  // Append the JS tag to the Document Body.
+  loadScript = () => {
+    const scriptId = 'arkose-script';
+    const currentScript = document.getElementById(scriptId);
+    if (currentScript) {
+      currentScript.remove();
+    }
+    const script = document.createElement('script');
+    script.id = scriptId;
+    script.type = 'text/javascript';
+    script.src = `https://client-api.arkoselabs.com/v2/${this.props.publicKey}/api.js`;
+    script.setAttribute('data-callback', 'setupEnforcement');
+    script.async = true;
+    script.defer = true;
+    if (this.props.nonce) {
+      script.setAttribute('data-nonce', this.props.nonce);
+    }
+    document.body.appendChild(script);
+    return script;
+  };
+
+  setupEnforcement = (myEnforcement) => {
+    this.myEnforcement = myEnforcement;
+    this.myEnforcement.setConfig({
+      selector: this.props.selector && `#${this.props.selector}`,
+      mode: this.props.mode,
+      onReady: () => {
+        if (this.props.onReady) {
```

Comment:
if we set up default props for all of the callbacks we don't need to do the exist checks in each callback and these could be as simple as

```onReady: this.props.onReady```